reps/2023-10-13-accelerator-support.md

```diff
@@ -0,0 +1,234 @@
+# Ray Accelerator Support
+
+## Summary
+
+Holistic design of supporting accelerators in Ray.
+This also serves as the reference doc for contributors
+to add new accelerator support in the future.
+
+### General Motivation
+
+Nowadays, more and more specialized accelerators (e.g. TPU, HPU) are coming out to speed up AI workloads.
+Supporting those accelerators natively in Ray will increase Ray's adoption as the computing framework for scaling AI applications.
+
+### Should this change be within `ray` or outside?
+
+Inside `ray` project since we want to support those accelerators out of the box.
+
+## Stewardship
+
+### Required Reviewers
+The proposal will be open to the public, but please suggest a few experienced Ray contributors in this technical domain whose comments will help this proposal. Ideally, the list should include Ray committers.
+
+@pcmoritz, @ericl, @scv119
+
+### Shepherd of the Proposal (should be a senior committer)
+To make the review process more productive, the owner of each proposal should identify a **shepherd** (should be a senior Ray committer). The shepherd is responsible for working with the owner and making sure the proposal is in good shape (with necessary information) before marking it as ready for broader review.
+
+@scv119
+
+## Design
+
+### Ray Core
+
+#### API
+Users will use accelerators through Ray resources. For GPU accelerators (e.g. Nvidia, AMD, Intel GPUs), they will use the `GPU` resource name. For other accelerators, they will each have their own resource name (e.g. TPU accelerator has resource name `TPU`). `accelerator_type` option can also be used to specify a particular type of accelerator for a task or actor.
+
+```python
+@ray.remote(num_gpus=1)
+def task():
+    ...
+
+@ray.remote(num_gpus=1, accelerator_type="A100")
+def task():
+    ...
+
+@ray.remote(num_gpus=1, accelerator_type="Intel-Max-1550")
+def task():
+    ...
+
+@ray.remote(resources={"TPU": 1})
+def task():
+    ...
+
+@ray.remote(resources={"TPU": 1}, accelerator_type="TPU-V4")
+def task():
+    ...
+```
+
+##### Alternatives
```

Comment:
For the alternatives, can you briefly describe why we went with the above instead of them?

Reply:
Added.